#!/usr/bin/env python
PACKAGE = "hydrus"

from dynamic_reconfigure.parameter_generator_catkin import *

LQI_GAIN_FLAG = 0
LQI_RP_P_GAIN = 1
LQI_RP_I_GAIN = 2
LQI_RP_D_GAIN = 3
LQI_Y_P_GAIN = 4
LQI_Y_I_GAIN = 5
LQI_Y_D_GAIN = 6
LQI_Z_P_GAIN = 7
LQI_Z_I_GAIN = 8
LQI_Z_D_GAIN = 9


gen = ParameterGenerator()

gen.add("lqi_gain_flag", bool_t, LQI_GAIN_FLAG, "LQI Gain Flag", False)

gen.add("q_roll", double_t, LQI_RP_P_GAIN, "LQI Roll and Pitch P Gain", 200, 0, 1000)
gen.add("q_roll_i", double_t, LQI_RP_I_GAIN, "LQI Roll and Pitch I Gain", 10, 0, 50)
gen.add("q_roll_d", double_t, LQI_RP_D_GAIN, "LQI Roll and Pitch D Gain", 20, 0, 300)

gen.add("q_yaw", double_t, LQI_Y_P_GAIN, "LQI Yaw P Gain", 1, 0, 1000)
gen.add("q_yaw_i", double_t, LQI_Y_I_GAIN, "LQI Yaw I Gain", 1, 0, 20)
gen.add("q_yaw_d", double_t, LQI_Y_D_GAIN, "LQI Yaw D Gain", 10, 0, 1000)

gen.add("q_z", double_t, LQI_Z_P_GAIN, "LQI Z P Gain", 1, 0, 100)
gen.add("q_z_i", double_t, LQI_Z_I_GAIN, "LQI Z I Gain", 1, 0, 20)
gen.add("q_z_d", double_t, LQI_Z_D_GAIN, "LQI Z D Gain", 10, 0, 200)

exit(gen.generate(PACKAGE, "hydrus", "LQI"))
